tests/vmi_multus_test.go

```diff
@@ -1010,59 +1015,98 @@ func cidrToIP(cidr string) string {
+	if vmi == nil {
+		return fmt.Errorf("the VMI must exist for us to configure an interface in it")
+	}
+
+	err, interfaceName := getSRIOVInterfaceNameByMAC(virtClient, vmi, interfaceMac)
+	if err != nil {
+		return fmt.Errorf("should get sriov interface: %w", err)
+	}
+
+	return configInterface(vmi, interfaceName, interfaceAddress)
+}
+
+func checkInterface(vmi *v1.VirtualMachineInstance, interfaceName string) error {
+	if vmi == nil {
+		return fmt.Errorf("the VMI must exist for us to check an interface in it")
+	}
+
+	cmdCheck := fmt.Sprintf("ip link show %s\n", interfaceName)
+	if err := runSafeCommand(vmi, cmdCheck); err != nil {
+		return fmt.Errorf("Interface %s was not found in the VMI %s within the given timeout: %w", interfaceName, vmi.Name, err)
+	}
+
+	return nil
 }
 
-func checkInterface(vmi *v1.VirtualMachineInstance, interfaceName string) {
+func checkMacAddress(vmi *v1.VirtualMachineInstance, interfaceName, macAddress string) error {
 	cmdCheck := fmt.Sprintf("ip link show %s\n", interfaceName)
-	err := console.SafeExpectBatch(vmi, []expect.Batcher{
+	if err := console.SafeExpectBatch(vmi, []expect.Batcher{
 		&expect.BSnd{S: "\n"},
 		&expect.BExp{R: console.PromptExpression},
 		&expect.BSnd{S: cmdCheck},
-		&expect.BExp{R: console.PromptExpression},
+		&expect.BExp{R: macAddress},
 		&expect.BSnd{S: "echo $?\n"},
 		&expect.BExp{R: console.RetValue("0")},
-	}, 15)
-	Expect(err).ToNot(HaveOccurred(), "Interface %q was not found in the VMI %s within the given timeout", interfaceName, vmi.Name)
+	}, 15); err != nil {
+		return fmt.Errorf("MAC %s was not found in the VMI %s within the given timeout: %w", macAddress, vmi.Name, err)
+	}
+
+	return nil
 }
 
-func checkMacAddress(vmi *v1.VirtualMachineInstance, interfaceName, macAddress string) {
-	cmdCheck := fmt.Sprintf("ip link show %s\n", interfaceName)
-	err := console.SafeExpectBatch(vmi, []expect.Batcher{
+func setUpInterface(vmi *v1.VirtualMachineInstance, interfaceName string) error {
+	setUpCmd := fmt.Sprintf("ip link set %s up\n", interfaceName)
+	if err := runSafeCommand(vmi, setUpCmd); err != nil {
+		return fmt.Errorf("failed to set interface %s up on VMI %s: %w", interfaceName, vmi.Name, err)
+	}
+
+	return nil
+}
+
+func runSafeCommand(vmi *v1.VirtualMachineInstance, command string) error {
+	return console.SafeExpectBatch(vmi, []expect.Batcher{
 		&expect.BSnd{S: "\n"},
 		&expect.BExp{R: console.PromptExpression},
-		&expect.BSnd{S: cmdCheck},
-		&expect.BExp{R: macAddress},
+		&expect.BSnd{S: command},
+		&expect.BExp{R: console.PromptExpression},
 		&expect.BSnd{S: "echo $?\n"},
 		&expect.BExp{R: console.RetValue("0")},
 	}, 15)
-	Expect(err).ToNot(HaveOccurred(), "MAC %q was not found in the VMI %s within the given timeout", macAddress, vmi.Name)
+}
+
+func getSRIOVInterfaceNameByMAC(virtClient kubecli.KubevirtClient, vmi *v1.VirtualMachineInstance, mac string) (error, string) {
```

Comment:
please consider reorder the return values,
also i am not sure we need here 2 , you can just return nil if not found and then assert its Not Empty

Reply:
> please consider reorder the return values,
Done.
> also i am not sure we need here 2 , you can just return nil if not found and then assert its Not Empty
I think that it could be useful to know what exactly failed. Could help with debugging. Just returning an empty string will provide less  information I think.

Reply:
there is just one failure there, so the error is trivial, imho we can keep it simpler

Reply:
I removed the GET from there, so you are right.
Done.